lib/arel/visitors/sqlserver.rb

```diff
@@ -11,13 +11,18 @@ class SQLServer < Arel::Visitors::ToSql
 
       private
 
-      # SQLServer ToSql/Visitor (Overides)
+      # SQLServer ToSql/Visitor (Overrides)
 
-      def visit_Arel_Nodes_BindParam o, collector
-        collector.add_bind(o.value) { |i| "@#{i - 1}" }
+      BIND_BLOCK = proc { |i| "@#{i - 1}" }
+      private_constant :BIND_BLOCK
+
+      def bind_block; BIND_BLOCK; end
```

Comment:
Do we really have to redefine `bind_block` method?

Reply:
Yes, you need to redefine it unfortunately.

```ruby
class Car
  COLOUR = 'white'

  def colour
    COLOUR
  end
end

class Ferrari < Car
  COLOUR = 'red'
end
```

```
Car.new.colour
=> white
puts Ferrari.new.colour
=> white
```